src/content/blog/how-to-configure-astropaper-theme.md
---
author: 'A. Kufta'
pubDatetime: 2022-09-23T04:58:53Z
modDatetime: 2024-01-15T13:05:56.066Z
title: How to configure AstroPaper theme
slug: how-to-configure-astropaper-theme
featured: true
draft: false
tags:
  - configuration
  - docs
description: How you can make AstroPaper theme absolutely yours.
---

AstroPaper is a highly customizable Astro blog theme. With AstroPaper, you can customize everything according to your personal taste. This article will explain how you can make some customizations easily in the config file.

## Table of contents

## Configuring SITE

The important configurations lies in `src/config.ts` file. Within that file, you'll see the `SITE` object where you can specify your website's main configurations.

During deveopment, it's okay to leave `SITE.website` empty. But in production mode, you should specify your deployed url in `SITE.website` option since this will be used for canonical URL, social card URL etc.. which are important for SEO.

```js
// file: src/config.ts
export const SITE = {
  website: "https://astro-paper.pages.dev/",
  author: "Sat Naing",
  desc: "A minimal, responsive and SEO-friendly Astro blog theme.",
  title: "AstroPaper",
  ogImage: "astropaper-og.jpg",
  lightAndDarkMode: true,
  postPerPage: 3,
  scheduledPostMargin: 15 * 60 * 1000, // 15 minutes
};
```

Here are SITE configuration options

| Options               | Description                                                                                                                                                                                                                                         |
| --------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `website`             | Your deployed website url                                                                                                                                                                                                                           |
| `author`              | Your name                                                                                                                                                                                                                                           |
| `desc`                | Your site description. Useful for SEO and social media sharing.                                                                                                                                                                                     |
| `title`               | Your site name                                                                                                                                                                                                                                      |
| `ogImage`             | Your default OG image for the site. Useful for social media sharing. OG images can be an external image url or they can be placed under `/public` directory.                                                                                        |
| `lightAndDarkMode`    | Enable or disable `light & dark mode` for the website. If disabled, primary color scheme will be used. This option is enabled by default.                                                                                                           |
| `postPerPage`         | You can specify how many posts will be displayed in each posts page. (eg: if you set SITE.postPerPage to 3, each page will only show 3 posts per page)                                                                                              |
| `scheduledPostMargin` | In Production mode, posts with a future `pubDatetime` will not be visible. However, if a post's `pubDatetime` is within the next 15 minutes, it will be visible. You can set `scheduledPostMargin` if you don't like the default 15 minutes margin. |

## Configuring locale

You can configure the default locale used for the build (e.g., date format in the post page), and for the rendering in browsers (e.g., date format in the search page)

```js
// file: src/config.ts
export const LOCALE = {
  lang: "en", // html lang code. Set this empty and default will be "en"
  langTag: ["en-EN"], // BCP 47 Language Tags. Set this empty [] to use the environment default
} as const;
```

`LOCALE.lang` will be used as HTML ISO Language code in `<html lang="en">`. If you don't specify this, default fallback will be set to `en`.
`LOCALE.langTag` is used as [datetime locale](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString#locales). For this, you can specify an array of locales for fallback languages. Leave `LOCALE.langTag` empty `[]` to use the environment default at _build-_ and _run-time_.

## Configuring logo or title

You can specify site's title or logo image in `src/config.ts` file.

![An arrow pointing at the website logo](https://res.cloudinary.com/noezectz/v1663911318/astro-paper/AstroPaper-logo-config_goff5l.png)

```js
// file: src/config.ts
export const LOGO_IMAGE = {
  enable: false,
  svg: true,
  width: 216,
  height: 46,
};
```

If you specify `LOGO_IMAGE.enable` => `false`, AstroPaper will automatically convert `SITE.title` to the main site text logo.

If you specify `LOGO_IMAGE.enable` => `true`, AstroPaper will use the logo image as the site's main logo.

You have to specify `logo.png` or `logo.svg` under `/public/assets` directory. Currently, only svg and png image file formats are supported. (**_Important!_** _logo name has to be logo.png or logo.svg)_

If your logo image is png file format, you have to set `LOGO_IMAGE.svg` => `false`.

It is recommended that you specify width and height of your logo image. You can do that by setting `LOGO_IMAGE.width` _and_ `LOGO_IMAGE.height`

## Configuring social links

You can configure your own social links along with its icons.

![An arrow pointing at social link icons](https://res.cloudinary.com/noezectz/v1663914759/astro-paper/astro-paper-socials_tkcjgq.png)

Currently 20 social icons are supported. (Github, LinkedIn, Facebook etc.)

You can specify and enable certain social links in hero section and footer. To do this, go to `/src/config.ts` and then you'll find `SOCIALS` array of object.

```js
// file: src/config.ts
export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: ` ${SITE.title} on Github`,
    active: true,
  },
  {
    name: "Facebook",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Facebook`,
    active: true,
  },
  {
    name: "Instagram",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Instagram`,
    active: true,
  },
  ...
]
```

You have to set specific social link to `active: true` in order to appear your social links in hero and footer section. Then, you also have to specify your social link in `href` property.

For instance, if I want to make my Github appear, I'll make it like this.

```js
export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/satnaing", // update account link
    linkTitle: `${SITE.title} on Github`, // this text will appear on hover and VoiceOver
    active: true, // makre sure to set active to true
  }
  ...
]
```

Another thing to note is that you can specify the `linkTitle` in the object. This text will display when hovering on the social icon link. Besides, this will improve accessibility and SEO. AstroPaper provides default link title values; but you can replace them with your own texts.

For example,

```js
linkTitle: `${SITE.title} on Twitter`,
```

to

```js
linkTitle: `Follow ${SITE.title} on Twitter`;
```

## Conclusion

This is the brief specification of how you can customize this theme. You can customize more if you know some coding. For customizing styles, please read [this article](https://astro-paper.pages.dev/posts/customizing-astropaper-theme-color-schemes/). Thanks for reading.✌🏻


src/content/blog/masterpost-of-awesome-resources-for-creating-beautiful-landscapes-and-environments.md
---
title: Awesome resources for creating beautiful landscapes and environments
author: 'A. Kufta'
pubDatetime: 2017-12-18T21:56:14+00:00
slug: awesome-resources-for-creating-beautiful-landscapes-and-environments
featured: false
tags:
  - masterpost
  - art theory
ogImage: ""
description: A masterpost of  30 + resources to help you draw and paint beautiful landscapes, environments, and backgrounds.
---
Learning to draw beautiful, expressive backgrounds, environments, landscapes is a wonderful skill that will help you become a more well rounded artist. This skill requires you to understand how different components integrate in space. This means sharpening up those **composition, perspective, color theory, and value skills**. 

It's a lot, but it's worth it. 



If you plan to create digitally, before you begin, please consider checking out a  [**Beginner's Guide to Digital Art**](https://artres.xyz/post/beginners-guide-to-digital-art/). The websites at the bottom are particularly useful for all artists! 

## Recommended Reading – Articles

* [This article ](https://t.umblr.com/redirect?z=http%3A%2F%2Ffav.me%2Fd8d2kin&t=MDYwZWI3MjhiMDQyYjMyMGIwN2VjNGEzZWEzZjIxMTk1NWNmMDdlZSxmcEh0S1RlMw%3D%3D&b=t%3Ac4za80VKBG4D5SwKeVtkjQ&p=http%3A%2F%2Fart-res.tumblr.com%2Fpost%2F137846311773%2Ftutorial-masterpost-on-backgrounds-middlegrounds&m=0)by [Goodnight-Melbourne](https://t.umblr.com/redirect?z=http%3A%2F%2Fgoodnight-melbourne.deviantart.com%2F&t=ZGEyMzU1YmM4ZjYwYTE5NmI5ZDNkMjFjNjE1Y2VjMmVkNWZjYjFjMSxmcEh0S1RlMw%3D%3D&b=t%3Ac4za80VKBG4D5SwKeVtkjQ&p=http%3A%2F%2Fart-res.tumblr.com%2Fpost%2F137846311773%2Ftutorial-masterpost-on-backgrounds-middlegrounds&m=0) provides an indepth tutorial on painting backgrounds. 
* [Awesome applied demo on depth and foregrounds, middle grounds, and backgrounds.](https://t.umblr.com/redirect?z=http%3A%2F%2Fdesign.tutsplus.com%2Ftutorials%2Fcreate-a-middle-earth-inspired-landscape-in-photoshop--psd-23767&t=MzMwNTkzNmI1MzZhNzAxMWRhOTllMzNiYTg4YmE3YzY5NjA5NTJmYSxmcEh0S1RlMw%3D%3D&b=t%3Ac4za80VKBG4D5SwKeVtkjQ&p=http%3A%2F%2Fart-res.tumblr.com%2Fpost%2F137846311773%2Ftutorial-masterpost-on-backgrounds-middlegrounds&m=0)
* [Covers composition in photography, but applies everywhere. ](https://t.umblr.com/redirect?z=http%3A%2F%2Fphotodoto.com%2Flearning-composition-foreground-middleground-background%2F&t=NGNhOWZiMWZlYTY3Yzk1OWMwMzY5ZjkyMzE0ZTE2YWFkNWJlZDYxNCxmcEh0S1RlMw%3D%3D&b=t%3Ac4za80VKBG4D5SwKeVtkjQ&p=http%3A%2F%2Fart-res.tumblr.com%2Fpost%2F137846311773%2Ftutorial-masterpost-on-backgrounds-middlegrounds&m=0)
* [Unifying Foregrounds and Backgrounds (TUTORIAL) ](https://alex-chow.deviantart.com/art/Unifying-Foregrounds-and-Backgrounds-TUTORIAL-550025320)by [Alex-Chow](https://alex-chow.deviantart.com/)
* [ Layout **Background** Exercise](https://art-res.tumblr.com/post/161516829610/layout-background-exercise)
* [Tips for Drawing Backgrounds ](https://kotaku.com/tips-for-drawing-backgrounds-1759168924)
* [How To Draw Landscapes For The Absolute Beginner ](https://www.drawing-pencil-sketches.com/how-to-draw-landscapes.html)
* [How to Draw : Drawing and Sketching Objects and Environments](https://www.basheergraphic.com/products/how-to-draw-drawing-and-sketching-objects-and-environments)
* [How to Draw With Perspective ](https://design.tutsplus.com/tutorials/how-to-draw-with-perspective--psd-7508)
* [Craft a Dramatic Vector Landscape Environment](https://design.tutsplus.com/tutorials/craft-a-dramatic-vector-landscape-environment--vector-24)
* [Improve Your Artwork by Learning to See Light and Shadow](https://design.tutsplus.com/articles/improve-your-artwork-by-learning-to-see-light-and-shadow--cms-20282)

## Videos & Channels

{{< youtube VkAOpEZQo20 >}}
{{< youtube P388z-u6zjY >}}
{{< youtube ujiwVregcTg >}}
{{< youtube XUVM9tgN2h4 >}}
{{< youtube uPD9lNj01V4 >}}
* <a href="https://www.youtube.com/results?search_query=3d+environment"> SketchUp Videos</a>
  * 3D should help you visualize and understand what you're trying to create in 3D. I actually recommend downloading SketchUp and experimenting. 
* [Mead McLean's Youtube](https://www.youtube.com/user/theausquared) - wonderful and detailed videos. Surprised he doesn't have more subscribers.

## Specific Tutorials 

Decided against including images to optimize performance. I picked these out by hand and most of them are semi-realistic to realistic and include theory as well as step by step. 

* [Background Tutorial](https://tanaw.deviantart.com/art/Background-Tutorial-431500070) by [tanaw](https://tanaw.deviantart.com/)
* [Background Tutorial 1](https://tervola.deviantart.com/art/Background-Tutorial-1-405300536)& [part2](https://tervola.deviantart.com/art/Background-Tutorial-2-405302000) by [Tervola](https://tervola.deviantart.com/)
* [Background Tutorial Part 1](https://nika-wolfgirl.deviantart.com/art/Background-tutorial-Part-1-208346837)  & [Background tutorial - Part 2](https://kalambo.deviantart.com/art/Background-tutorial-Part-2-255819198) by [kalambo](https://kalambo.deviantart.com/)
* [Background Tutorial](https://snowfirewolf.deviantart.com/art/Background-Tutorial-331270204) by [Snowfirewolf](https://snowfirewolf.deviantart.com/)
* [Background Art Process/Tutorial](https://sendrawz.deviantart.com/art/Background-Art-Process-Tutorial-613416167) by [sendrawz](https://sendrawz.deviantart.com/)
* \[[Step Art\] Background Tutorial](https://sakon04.deviantart.com/art/Step-Art-Background-Tutorial-435708648) by [Sakon04](https://sakon04.deviantart.com/) - Walkthrough / Step by Step Demo
* [Landscape/Background Tutorial](https://seerlight.deviantart.com/art/Landscape-Background-Tutorial-693664472)by [SeerLight](https://seerlight.deviantart.com/)

Hope you enjoyed this post! Consider sharing this post if it helped! ❤️

src/config.ts
import type { Site, SocialObjects } from "./types";

export const SITE: Site = {
  website: "http://localhost:4321", // replace this with your deployed domain
  author: "Allison Kufta",
  desc: "A minimal, responsive and SEO-friendly Astro blog theme.",
  title: "Art-Res",
  ogImage: "astropaper-og.jpg",
  lightAndDarkMode: true,
  postPerPage: 10,
  scheduledPostMargin: 15 * 60 * 1000, // 15 minutes
};

export const LOCALE = {
  lang: "en", // html lang code. Set this empty and default will be "en"
  langTag: ["en-EN"], // BCP 47 Language Tags. Set this empty [] to use the environment default
} as const;

export const LOGO_IMAGE = {
  enable: true,
  svg: true,
  width: 70,
  height: 70,
};

export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/akufta",
    linkTitle: ` ${SITE.title} on Github`,
    active: true,
  },
  {
    name: "Facebook",
    href: "https://www.facebook.com/helloartres/",
    linkTitle: `${SITE.title} on Facebook`,
    active: true,
  },
  {
    name: "Instagram",
    href: "https://instagram.com/akufta.art",
    linkTitle: `${SITE.title} on Instagram`,
    active: true,
  },
  {
    name: "LinkedIn",
    href: "",
    linkTitle: `${SITE.title} on LinkedIn`,
    active: false,
  },
  {
    name: "Mail",
    href: "mailto:aky@artres.xyz",
    linkTitle: `Send an email to ${SITE.title}`,
    active: true,
  },
  {
    name: "Twitter",
    href: "https://www.twitter.com/allisonkufta",
    linkTitle: `${SITE.title} on Twitter`,
    active: true,
  },
  {
    name: "Bluesky",
    href: "https://bsky.app/profile/akufta.bsky.social",
    linkTitle: `${SITE.title} on Bluesky`,
    active: true,
  },
  {
    name: "Twitch",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Twitch`,
    active: false,
  },
  {
    name: "YouTube",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on YouTube`,
    active: false,
  },
  {
    name: "WhatsApp",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on WhatsApp`,
    active: false,
  },
  {
    name: "Snapchat",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Snapchat`,
    active: false,
  },
  {
    name: "Pinterest",
    href: "https://www.pinterest.com/astrikosart/",
    linkTitle: `${SITE.title} on Pinterest`,
    active: true,
  },
  {
    name: "TikTok",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on TikTok`,
    active: false,
  },
  {
    name: "CodePen",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on CodePen`,
    active: false,
  },
  {
    name: "Discord",
    href: "https://discord.gg/cw7EdPg",
    linkTitle: `${SITE.title} on Discord`,
    active: true,
  },
  {
    name: "GitLab",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on GitLab`,
    active: false,
  },
  {
    name: "Reddit",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Reddit`,
    active: false,
  },
  {
    name: "Skype",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Skype`,
    active: false,
  },
  {
    name: "Steam",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Steam`,
    active: false,
  },
  {
    name: "Telegram",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Telegram`,
    active: false,
  },
  {
    name: "Mastodon",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Mastodon`,
    active: false,
  },
];
